### ISSUE TITLE
Error when converting boolean column to decimal in PostgreSQL 

### ISSUE THREAD

**Issue by shemgp** [laravel/framework#28963]
- Laravel Version: 5.8.26
- PHP Version: 7.3.6
- Database Driver & Version: PostgreSQL 11.4

### Description:
Converting boolean to decimal/numeric in migrations results in:

```
Datatype mismatch: 7 ERROR:  column "${column_name}" cannot be cast automatically to type numeric
```

### Steps To Reproduce:
1. Create migration with a boolean column. Example:

```php
$table->boolean('special_price')->nullable();
```
2. Create another migration that will alter the column to make it decimal. Example:

```php
$table->decimal('special_price', 12, 4)->nullable()->change();
```
3. Run `php artisan migrate` and it will show an error like above. More complete error message:
```
   Illuminate\Database\QueryException  : SQLSTATE[42804]: Datatype mismatch: 7 ERROR:  column "special_price" cannot be cast automatically to type numeric
HINT:  You might need to specify "USING special_price::numeric(12,4)". (SQL: ALTER TABLE product_flat ALTER special_price TYPE NUMERIC(12, 4))
```

Thanks for your help.

**Comment by shemgp** [comment]
Here are sample complete migration files that will show the error:
[migrations.zip](https://github.com/laravel/framework/files/3330690/migrations.zip)

**Comment by staudenmeir** [comment]
Does the table already contain data?

**Comment by mfn** [comment]
This is currently expected: you need to explicitly tell the SQL statement _how_ to convert it.

Postgres itself wouldn't know what `false` or `true` should be represented in `DECIMAL(12,4)`.

Laravel doesn't currently support a way to express this; you need to use a raw SQL statement and formulate your transformation.

**Comment by shemgp** [comment]
@staudenmeir no, it's a fresh install.

@mfn Perhaps there's a way of adding, to the postgresql grammar, to automatically add something like `USING special_price::numeric(12,4)`? to the SQL statement?

**Comment by staudenmeir** [comment]
Adding `USING special_price::numeric(12,4)` doesn't fix the issue, the query still fails.

You'll need to drop the column and add it again:

```php
Schema::table('product_flat', function (Blueprint $table) {
    $table->dropColumn('special_price');
});

Schema::table('product_flat', function (Blueprint $table) {
    $table->decimal('special_price', 12, 4)->nullable();
});
```

**Comment by shemgp** [comment]
Well, I guess my request would be that there be a `->using()` function? Since changes like this would really need custom `USING`: 
```sql
ALTER TABLE product_flat ALTER special_price TYPE numeric (12,4) USING special_price::int::numeric(12,4);
```

**Comment by staudenmeir** [comment]
I don't think this is feasible: The queries for altering tables are generated but by the `doctrine/dbal` package, not by Laravel.

This feature has been suggested: https://github.com/doctrine/dbal/issues/2376

**Comment by laurencei** [comment]
Closing as it belongs to another package...

**Comment by mfn** [comment]
> You'll need to drop the column and add it again:

This is technically not entirely correct; _especially_ if you've existing data you want to preserve.

As we've established, Postgres misses infrastructure in the schema builder to express using, but certainly you can _convert_ it without destroying, using raw SQL as I initially suggested.

Just because Postgres doens't know how to _automatically_ convert it, you just have to specify the transformation yourself.

Example:
```sql
ALTER TABLE table_name ALTER COLUMN special_price TYPE DECIMAL(12,4) USING CASE WHEN special_price = false THEN 0 ELSE 1 END;
```

But, I agree, this shows that it's not feasible to implement in the builder as the syntax can get even more complex than this.

**Comment by shemgp** [comment]
@staudenmeir @laurencei This works: https://github.com/laravel/framework/compare/5.8...shemgp:5.8 

Do you think it'll be accepted?

**Comment by laurencei** [comment]
@shemgp - given you've already done the code - open a PR. Taylor will review and decide.

My suggestion is target `master` so it'll be included in 5.9 which is due out soon.

**Comment by xyrintech** [comment]
Any update for this?
